src/material/progress-spinner/progress-spinner.ts

```diff
@@ -218,6 +222,14 @@ export class MatProgressSpinner extends _MatProgressSpinnerBase implements OnIni
         this.strokeWidth = defaults.strokeWidth;
       }
     }
+
+    // Safari has an issue where the circle isn't positioned correctly when the page has a
+    // different zoom level from the default. This handler triggers a recalculation of the
+    // `transform-origin` when the page zoom level changes.
+    // See `_getCircleTransformOrigin` for more info.
+    if (_platform.isBrowser && _platform.SAFARI) {
+      window.addEventListener('resize', this._resizeHandler);
```

Comment:
I'm not a fan of this, but I couldn't find a better way to trigger a change detection when the zoom changes. Alternatively we could not trigger the change detection ourselves, but we risk having the user see a broken animation when they zoom in/out.